src/hera/workflows/_meta_mixins.py

```diff
@@ -539,8 +545,7 @@ def _create_subnode(
         assert _context.pieces
 
         template_ref = None
-        _context.declaring = False
-        if _context.pieces[0] != self and isinstance(self, WorkflowTemplate):
+        if _context.pieces[0] is not self and isinstance(self, WorkflowTemplate):
```

Comment:
Note: `!=` in Pydantic raises an "incorrect type" validation error due to having been returned a string from TemplateInvocatorSubNodeMixin's modified `__getattribute__` implementation. We could temporarily disable the active context instead, but switching to `is not` is simpler (and faster). As a side-benefit, we can remove the temporary assignment of `False` to `_context.declaring`, which was previously working around this issue.